### DISCUSSION TITLE
How to make sub slash command in cogs

### DISCUSSION THREAD

**Discussion by Jourdelune** [Pycord-Development/pycord#251] [Q&A] [answered]
Hi, I try to make sub slash commands in cogs with this code:
```py
from os import name

from discord.app import slash_command, Option
from discord.app import SlashCommandGroup
from googletrans import Translator
from discord.ext import commands
import discord

from Function import *

translator = Translator()

class Config(commands.Cog):
    def __init__(self, bot):
        self.bot = bot
    
    math = commands.Bot.command_group(name="math", description="Commands related to mathematics.") 

    @math.command()
    async def add(ctx, num1: int, num2: int):
        """Get the sum of 2 integers."""
        await ctx.respond(f"The sum of these numbers is **{num1+num2}**")


def setup(bot): 
    
    
    math = SlashCommandGroup(name="math", description="Commands related to mathematics.")
    bot.add_cog(Config(bot))
```
But I can't set math (`math = commands.Bot.command_group(name="math", description="Commands related to mathematics.") `).

**Comment by Dorukyum** [comment, marked as answer]
We are currently working on subcommands in cogs, so they arent ready to use yet.
Also your usage would be wrong anyway. I suggest learning about Object Oriented Programming with Python before making a Discord bot.

**Reply by Jourdelune** [reply]
It is impossible to do at the moment?

**Reply by Dorukyum** [reply]
You can join the Discord server to stay updated.

**Comment by notPronoe** [comment]
For anyone stumbling onto this discussion from google, this has now been implemented. c.f. https://github.com/Pycord-Development/pycord/blob/master/examples/app_commands/slash_cog_groups.py
